pkg/block/block.go

```diff
@@ -126,6 +128,33 @@ func cleanUp(logger log.Logger, bkt objstore.Bucket, id ulid.ULID, err error) er
 	return err
 }
 
+// MarkForDeletion creates a file which stores information about when the block was marked for deletion.
+func MarkForDeletion(ctx context.Context, logger log.Logger, bkt objstore.Bucket, id ulid.ULID) error {
+	deletionMarkExists, err := objstore.Exists(ctx, bkt, path.Join(id.String(), metadata.DeletionMarkFilename))
+	if err != nil {
+		return errors.Wrapf(err, "check compactor meta for id %s in bucket", id.String())
+	}
+	if deletionMarkExists {
+		return nil
+	}
+
+	deletionMark, err := json.Marshal(metadata.DeletionMark{
+		ID:           id,
+		DeletionTime: time.Now().Unix(),
+		Version:      2,
+	})
+	if err != nil {
+		return errors.Wrap(err, "compactor meta json marshalling")
+	}
+
+	if err := bkt.Upload(ctx, path.Join(id.String(), metadata.DeletionMarkFilename), bytes.NewReader(deletionMark)); err != nil {
+		return errors.Wrap(err, "upload compactor-meta file to bucket")
+	}
+
+	level.Info(logger).Log("msg", "block", id, "has been marked for deletion")
```

Comment:
got it, I thought that it separates them out with spaces, so If I say `Log("v1", "v2")`, it would print as `v1 v2`, instead it prints as `v1=v2`. Thank you for the detailed explanation. I didn't know about this.

Reply:
`log.Println` from standard library works that way indeed.